### PR TITLE
update diagnosis node DICT-228

### PR DESCRIPTION
**Pull request by bullenc**

### CODE REVIEW

File: gdcdictionary/schemas/diagnosis.yaml [hunk 1]
```diff
@@ -1322,6 +1322,13 @@ properties:
         $ref:
           - "_terms_enum.yaml#/not_allowed_to_collect/common"
 
+  cancer_detection_method:
+    enum:
+      - Symptomatic
+      - Screening
+      - Surveillance
+      - Not Reported
+
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:1331]: Alphabetize these

File: gdcdictionary/schemas/diagnosis.yaml [hunk 2]
```diff
@@ -18046,6 +18091,23 @@ properties:
...
+        - Gastrosplenic ligament
+        - Brain, Cerebrum
+        - Brain, Frontal lobe
+        - Brain, Temporal lobe
+        - Brain, Parietal
+        - Brain, Occipital lobe
+        - Brain, Ventricle, NOS
+        - Brain, Cerebellum, NOS
+        - Brain, Brain stem
     enumDef:
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:18111]: Alphabetize enums.

File: gdcdictionary/schemas/diagnosis.yaml [hunk 3]
```diff
@@ -18063,6 +18125,9 @@ properties:
         $ref:
           - "_terms_enum.yaml#/not_reported/common"
 
+  sites_of_involvement_count:
+    type: integer
+
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:18130]: Add "min: 1" here?  @tlicht3

File: gdcdictionary/schemas/diagnosis.yaml [hunk 4]
```diff
@@ -19624,6 +19695,34 @@ properties:
...
+        - Mitotic Rate > 5/50 HPF
+        - Atypical Mitotic Figures
+        - Cytoplasm presence <= to 25%
+        - Diffuse Architecture
+        - Necrosis
+        - Venous Invasion
+        - Sinusoid Invasion
+        - Invasion of Tumor Capsule
+        - Not Reported
+
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:19725]: Alphabetize

File: gdcdictionary/schemas/diagnosis.yaml [hunk 5]
```diff
@@ -2084,6 +2108,7 @@ properties:
       - Undifferentiated
       - Unknown
       - Not Reported
+      - Undifferentiated or Poorly Differentiated
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:2111]: Alphabetize enum

File: gdcdictionary/schemas/diagnosis.yaml [hunk 6]
```diff
@@ -7145,6 +7170,26 @@ properties:
         $ref:
           - "_terms_enum.yaml#/type_2/common"
 
+  pediatric_kidney_staging:
+    enum:
+      - Nephrectomy specimen with tumor confined to the kidney; no distant metastasis
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:7175]: Do we have enums that span multiple lines in other places?  I'm okay with this as long as it works with submission and the dictionary viewer.

File: gdcdictionary/schemas/diagnosis.yaml [hunk 7]
```diff
@@ -19624,6 +19695,34 @@ properties:
...
+        - Atypical Mitotic Figures
+        - Cytoplasm presence <= to 25%
+        - Diffuse Architecture
+        - Necrosis
+        - Venous Invasion
+        - Sinusoid Invasion
+        - Invasion of Tumor Capsule
+        - Not Reported
+
   weiss_assessment_score:
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:19726]: tissue_or_organ_of_origin new enums seem to be missing

**Review by wwysoc2 — CHANGES_REQUESTED** [review]

File: gdcdictionary/schemas/diagnosis.yaml (hunk 6 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:7175]: https://codebeautify.org/yaml-editor-online I tried the object in the editor I pasted and it resolved to multi-line enums being placed onto on line each, sounds like to keep them multi-line we would actually need to do implement a parameter or flag

File: gdcdictionary/schemas/diagnosis.yaml (hunk 7 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:19726]: See my comment from 1/17 on diagnosis ticket (https://jira.opensciencedatacloud.org/browse/DICT-228), there were duplicate entries for upper/middle/lower third of esophagus caught by travis tests, i.e. values already exist in yaml file

File: gdcdictionary/schemas/diagnosis.yaml (hunk 5 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:2111]: done 👍

File: gdcdictionary/schemas/diagnosis.yaml (hunk 4 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:19725]: done 👍

File: gdcdictionary/schemas/diagnosis.yaml (hunk 3 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18130]: done 👍

Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18130]: Add minimum, can remove if decided not needed

File: gdcdictionary/schemas/diagnosis.yaml (hunk 2 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18111]: done 👍

File: gdcdictionary/schemas/diagnosis.yaml (hunk 1 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:1331]: done 👍

File: gdcdictionary/schemas/diagnosis.yaml [hunk 8]
```diff
@@ -2353,13 +2378,13 @@ properties:
...
       - "_terms.yaml#/last_known_disease_status/common"
     enum:
+      - Biochemical evidence of disease without structural correlate
       - Distant met recurrence/progression
       - Loco-regional recurrence/progression
-      - Biochemical evidence of disease without structural correlate
       - Tumor free
       - Unknown tumor status
       - With tumor
-      - not reported
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:2362]: Have we checked the migration status for this one?  Are these values already populated?

File: gdcdictionary/schemas/diagnosis.yaml [hunk 9]
```diff
@@ -16698,8 +16743,8 @@ properties:
     enum:
       - "yes"
       - "no"
-      - unknown
-      - not reported
+      - Unknown
+      - Not Reported
       - Not Allowed To Collect
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:16748]: Will this require deprecation-migration?

File: gdcdictionary/schemas/diagnosis.yaml [hunk 10]
```diff
@@ -16750,8 +16795,8 @@ properties:
...
```
Comment (wwysoc2) [inline gdcdictionary/schemas/diagnosis.yaml:16800]: Will this require deprecation-migration?

File: gdcdictionary/schemas/diagnosis.yaml (hunk 9 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:16748]: hmm not sure, good point. I'll revert the changes for this and other like it.

File: gdcdictionary/schemas/diagnosis.yaml [hunk 11]
```diff
@@ -18012,13 +18067,18 @@ properties:
         - Ovary, NOS
         - Ovary, Right
         - Pancreas
+        - Parametrium, left
+        - Parametrium, right
         - Parotid gland
         - Pelvic Peritoneum
+        - Pelvic lymph node(s)
```
Comment (wteouchicago) [inline gdcdictionary/schemas/diagnosis.yaml:18074]: This is "Pelvic Lymph Node(s)" in jira.

File: gdcdictionary/schemas/diagnosis.yaml [hunk 12]
```diff
@@ -18012,13 +18067,18 @@ properties:
...
         - Pelvis
+        - Peri-aortic lymph node(s)
```
Comment (wteouchicago) [inline gdcdictionary/schemas/diagnosis.yaml:18076]: this is "Peri-aortic Lymph Node(s)" in jira

File: gdcdictionary/schemas/diagnosis.yaml [hunk 13]
```diff
@@ -18012,13 +18067,18 @@ properties:
         - Ovary, NOS
         - Ovary, Right
         - Pancreas
+        - Parametrium, left
```
Comment (wteouchicago) [inline gdcdictionary/schemas/diagnosis.yaml:18070]: "Left" in jira

File: gdcdictionary/schemas/diagnosis.yaml [hunk 14]
```diff
@@ -18012,13 +18067,18 @@ properties:
...
+        - Parametrium, right
```
Comment (wteouchicago) [inline gdcdictionary/schemas/diagnosis.yaml:18071]: "Right" in jira

File: gdcdictionary/schemas/diagnosis.yaml [hunk 15]
```diff
@@ -17983,6 +18037,7 @@ properties:
         - Fallopian Tube, NOS
         - Fallopian Tube, Right
         - Gallbladder
+        - Gastrosplenic ligament
```
Comment (wteouchicago) [inline gdcdictionary/schemas/diagnosis.yaml:18040]: "Ligament" in jira

File: gdcdictionary/schemas/diagnosis.yaml (hunk 11 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18074]: Updated to what's in JIRA. It looks like past releases were not consistent with case of enums, will probably need to bring up in data dict meeting and discuss if a large set of migrations to be consistent will be needed...

File: gdcdictionary/schemas/diagnosis.yaml (hunk 13 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18070]: Updated to what's in JIRA.

File: gdcdictionary/schemas/diagnosis.yaml (hunk 15 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18040]: Updated to what's in JIRA.

File: gdcdictionary/schemas/diagnosis.yaml (hunk 14 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml:18071]: Updated to what's in JIRA.

File: gdcdictionary/schemas/diagnosis.yaml (hunk 12 above)
Comment (bullenc) [reply gdcdictionary/schemas/diagnosis.yaml]: Updated to what's in JIRA.

File: gdcdictionary/schemas/diagnosis.yaml (hunk 6 above)
Comment (wwysoc2) [reply gdcdictionary/schemas/diagnosis.yaml:7183]: Changed them to one line each.  We'll need to verify this looks okay in the viewer.

**Comment by wwysoc2** [comment]
pediatric_kidney_staging needs an entry in _terms.yaml.  @tlicht3  could you help with this?

File: gdcdictionary/schemas/diagnosis.yaml [hunk 16]
```diff
@@ -7145,6 +7176,20 @@ properties:
         $ref:
           - "_terms_enum.yaml#/type_2/common"
 
+  pediatric_kidney_staging:
+    $ref:
+      - "_terms.yaml#/pediatric_kidney_staging/common"
```
Comment (mpsolano) [inline gdcdictionary/schemas/diagnosis.yaml:7181]: I don't think there is a corresponding entry in `_terms.yaml` for `pediatric_kidney_staging` which may be the reason the build is failing

File: gdcdictionary/schemas/diagnosis.yaml (hunk 3 above)
Comment (tlicht3) [reply gdcdictionary/schemas/diagnosis.yaml:18132]: @wwysoc2 and @bullenc I think this is fine... I don't think folks should submit anything to this property if there is not involvement/no cancer.

File: gdcdictionary/schemas/diagnosis.yaml (hunk 6 above)
Comment (tlicht3) [reply gdcdictionary/schemas/diagnosis.yaml:7183]: @bullenc and @wwysoc2 thanks for looking into this.

File: gdcdictionary/schemas/diagnosis.yaml (hunk 16 above)
Comment (tlicht3) [reply gdcdictionary/schemas/diagnosis.yaml:7181]: @wwysoc2 and @mpsolano I think that's my bad - I can update that... was supposed to do that on Wednesday.

**Review by mpsolano — APPROVED** [review]

**Review by wwysoc2 — APPROVED** [review]

**Review by wwysoc2 — APPROVED** [review]
👍

**Pull request merged**